Added best effort support for Collapse trigger detection

phistuck opened the pull request:
The trigger was detected only if the target element had an ID attribute and it matched the value of either the href or data-target of the trigger.
This adds another minimal effort to the detection - if Collapse is triggered by clicking on a trigger, add it to the trigger list.
Test case -

``` html
<a data-toggle="collapse" data-target="[data-me=&quot;1&quot]">Trigger</a><div data-me="1" class="collapse"></div>
```

phistuck committed:
Added best effort support for Collapse trigger detection

The trigger was detected only if the target element had an ID attribute and it matched the value of either the href or data-target of the trigger.
This adds another minimal effort to the detection - if Collapse is triggered by clicking on a trigger, add it to the trigger list.
Test case -
<a data-toggle="collapse" data-target="[data-me=&quot;1&quot]">Trigger</a><div data-me="1" class="collapse"></div>

twbs-savage commented:
**Tests passed.** Automated cross-browser testing via Sauce Labs and Travis CI shows that the JavaScript changes in this pull request are: **CONFIRMED**

Commit: ebaf8c63e69a5b8d7ebe4539537573d75e212d43
Build details: https://travis-ci.org/twbs-savage/bootstrap/builds/69127510

(*Please note that this is a [fully automated](https://github.com/twbs/savage) comment.*)

cvrebert commented:
* Please fix the code style; see https://travis-ci.org/twbs/bootstrap/jobs/69127483#L424
* JavaScript changes require the addition of corresponding unit tests; see the readme on https://github.com/twbs/bootstrap/tree/master/js/tests for relevant how-to info.

phistuck committed:
Fixed style issues

Made Travis CI happy by fixing style nits.

twbs-savage commented:
**Tests passed.** Automated cross-browser testing via Sauce Labs and Travis CI shows that the JavaScript changes in this pull request are: **CONFIRMED**

Commit: d123dfc2b674f2c7b11df664fb98365bdadea0b9
Build details: https://travis-ci.org/twbs-savage/bootstrap/builds/69240353

(*Please note that this is a [fully automated](https://github.com/twbs/savage) comment.*)

phistuck committed:
Added unit tests for the improved trigger detection

Based on the existing tests, with an added non ID targeted trigger.

twbs-savage commented:
**Tests failed.** Automated cross-browser testing via Sauce Labs and Travis CI shows that the JavaScript changes in this pull request are: **BUSTED**

Commit: 9e8e94248f6ac379c7dff7dbae1b93a7d3413d93
Build details: https://travis-ci.org/twbs-savage/bootstrap/builds/69241704

(*Please note that this is a [fully automated](https://github.com/twbs/savage) comment.*)

phistuck committed:
Fixed a syntax error

&qupt; should be &quot;.

twbs-savage commented:
**Tests failed.** Automated cross-browser testing via Sauce Labs and Travis CI shows that the JavaScript changes in this pull request are: **BUSTED**

Commit: 730ca0c56cd68b62391fe507fb2190e9877fae52
Build details: https://travis-ci.org/twbs-savage/bootstrap/builds/69242079

(*Please note that this is a [fully automated](https://github.com/twbs/savage) comment.*)

phistuck committed:
Fixed the unit tests

- Changed the assert count to match the assert count of the tests.
- Renamed a test appropriately.

twbs-savage commented:
**Tests passed.** Automated cross-browser testing via Sauce Labs and Travis CI shows that the JavaScript changes in this pull request are: **CONFIRMED**

Commit: 5b6d55bb9399bd69601ada2c6db1a7cf5b55518b
Build details: https://travis-ci.org/twbs-savage/bootstrap/builds/69243424

(*Please note that this is a [fully automated](https://github.com/twbs/savage) comment.*)

phistuck commented:
@cvrebert - done.

phistuck commented:
@cvrebert - sorry, but are there any remaining issues?

phistuck commented:
@cvrebert - ping... :(

cvrebert commented:
CC: @fat for a design decision on whether we want to support usage of non-ID selectors

phistuck commented:
@fat - ping.

phistuck commented:
@cvrebert & @fat - ping again?

phistuck commented:
And another ping...
@cvrebert & @fat

cvrebert commented:
What with Bootstrap v4's release being so close and this being something of a nice-to-have rather than a must-have, I'm inclined to stop kicking the can down the road and just decline this. @phistuck Thanks for trying though; we appreciate your effort!

@twbs/team Feel free to reopen if you disagree.
